clasi_user/interface.py

```diff
@@ -0,0 +1,173 @@
+        '''
+        if not isinstance(path,(list,tuple)):
+            path = self._listify(path)     #if things are not iteratorable
+        self.path = self._clean_non_json_path(path) #remove non .json paths
+        self.csv_path = None
+        self.bs = bs
+        self.language_model_data = None
+        self.classification_model_data = None
+        self.eval_mode = eval_mode
+        self.df_train, self.df_valid = None, None
+        self.data_list, self.dataset_name = [], [] #to store all individual df for individual model
+
+    def _clean_non_json_path(self,path):
+        filenames = []
+        for file_path in path:
+            file_extension = file_path.name.split('.')
+            if len(file_extension) < 2: continue
+            else:
+                if file_path.name.split('.')[1] == 'json':
+                    filenames.append(file_path)
+        return filenames
+
+    def _listify(self,p):
+        '''Trun everying in p to a Path object'''
+
+        if p is None: p = []
+        elif isinstance(p,str): p = [p]
+        else:
+            if not isinstance(p,Iterable): p = [p]
+        return [Path(i) for i in p]
+
+    def _check_data_format(self):
+        if self.path == []: return False
+        one_file = self.path[0]
+        flag = 0
+        with open(one_file) as f:
+            data = json.load(f)
+        basilica_keys = ['data_points', 'num_data_points', 'id']
+        baslilca_data_keys = ['body', 'label', 'source', 'embedding']
+        for key in basilica_keys:
+            if not key in data.keys(): flag +=1
+        for key in baslilca_data_keys:
+            if not key in data['data_points'][0].keys(): flag +=1 #only check the first item...
+        if flag != 0:
+            print('Warning: data source not from Basilica, exit...')
+            return False
+        return True
+
+    def _convert_json_to_csv(self,test=False):
+        #use self.path to grab json format Basilica data
+        #return csv file
+        #body,label,source
+        body, label, source = [], [], []
+        for one_file_path in self.path:
+            with open(one_file_path) as file:
+                data = json.load(file)
+            for item in data['data_points']:
+                body.append(item['body'])
+                label.append(item['label'])
+                source.append(item['source'])
+            df = pd.DataFrame(list(zip(body, label, source)),
+                             columns = ['Body', 'Label', 'Source'])
+
+            #check if the parent directory exists
+            if test:
+                if not os.path.exists(one_file_path.parent/'delete_test'):
+                    os.makedirs(one_file_path.parent/'delete_test')
+            else:
+                if not os.path.exists(one_file_path.parent/'csv'):
+                    os.makedirs(one_file_path.parent/'csv')
+
+            if test: dest = one_file_path.parent/'delete_test'
+            else: dest = one_file_path.parent/'csv'
```

Comment:
It might be better to change this code block to the following:

```python
if test: dest = one_file_path.parent/'delete_test'
else: dest = one_file_path.parent/'csv'

if not os.path.exists(dest):
    os.makedirs(dest)
```